src/plugin/memory-webview-main.ts

```diff
@@ -141,20 +147,82 @@ export class MemoryWebview {
             this.messenger.onRequest(readMemoryType, request => this.readMemory(request), { sender: participant }),
             this.messenger.onRequest(writeMemoryType, request => this.writeMemory(request), { sender: participant }),
             this.messenger.onRequest(getVariables, request => this.getVariables(request), { sender: participant }),
+            this.messenger.onRequest(getMemoryDisplayConfigurationType, () => this.getMemoryDisplayConfiguration(), { sender: participant }),
+            this.messenger.onRequest(getColumnsVisibility, () => this.getColumnConfigurations(), { sender: participant }),
+            this.messenger.onNotification(setMemoryDisplayConfigurationType, request => this.setConfiguration(request), { sender: participant }),
+            this.messenger.onNotification(columnVisibilityType, request => this.handleColumnToggled(request), { sender: participant }),
```

Comment:
Is there a reason to handle these separately, or could the messages be combined with a payload that takes the same shape as the dendrified configuration we would get from `vscode.workspace.getConfiguration('memory-inspector')?

Reply:
I had separated them (maybe prematurely) since it felt like the ColumnContributionService had a separate set of responsibilities than the core MemoryWidget. Since columns are added somewhat dynamically, but we don't really have a way of dynamically adding new configuration options for each column, I decided to split the difference and separate out the preferences responsible for column visibitiliy. I'm happy to merge them into a single object if that feels better